### ISSUE TITLE
.NET designer throws an exception when project resource is serialized with System.Text.Json

### ISSUE THREAD

**Issue by devcsantos** [dotnet/winforms#12597]
### Environment

Visual Studio 17.12.3

### .NET version

.NET 8+

### Did this work in a previous version of Visual Studio and/or previous .NET release?

Yes, this previously worked on a previous Visual Studio 17.12 update.

### Issue description

We use `System.Text.Json` serialization to independently transfer data between processes. However, external action(adding System.Text.Json V9.0) breaks our code.

### Steps to reproduce

1. Download [DevExpress Installer](https://go.devexpress.com/DevexpressDownload_UniversalTrial.aspx) and install WinForms components.
2. Create a new DevExpress WinForms Project
3. Add the `System.Text.Json` NuGet package to the project.
4. Add SimpleButton
5. Invoke our Image Picker Dialog
6. Select a Vector Image or Font Icon and add it as a project resource.
Current result:
Exception is thrown

Attached short video that illustrates the issue and the project used.
[DXSample.zip](https://github.com/user-attachments/files/18033613/DXSample.zip)
[Video](https://github.com/user-attachments/assets/d25b474d-0a1a-4b51-94a0-698c19c3eada)

### Diagnostics

```text
System.MissingMethodException: Method not found: 'Int64 System.IO.Pipelines.PipeWriter.get_UnflushedBytes()'.
   at System.Text.Json.Serialization.JsonConverter.ShouldFlush(WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.DictionaryOfTKeyTValueConverter`3.OnWriteResume(Utf8JsonWriter writer, TCollection value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonDictionaryConverter`3.OnTryWrite(Utf8JsonWriter writer, TDictionary dictionary, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.Serialize(Utf8JsonWriter writer, T& rootValue, Object rootValueBoxed)
   at System.Text.Json.JsonSerializer.WriteString[TValue](TValue& value, JsonTypeInfo`1 jsonTypeInfo)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at DevExpress.Utils.Design.ResourceProjectHelperNetCore.WriteImportedResources(Dictionary`2 importedResources, Dictionary`2 resXFilePathsCore, Dictionary`2 existingResourcesCore, String selectedResource)
   at DevExpress.Utils.Design.SvgImageEditorViewModel.WriteImportedResourceAndLoadEditValue()
   at DevExpress.Utils.Design.SvgImageEditorViewModel.ImportResources(String[] fileNames)
   at DevExpress.Utils.Design.ImageEditorView.ImportResourceToProject(String[] fileNames)
   at DevExpress.Utils.Design.ImageEditorView.ImportToProject(DXImageGalleryItem[] items)
   at DevExpress.Utils.Design.ImagePickerGalleryBasePage.ApplyChanged()
   at DevExpress.Utils.Design.ImageEditorFormBase.ApplyChangesCore()
   at DevExpress.Utils.Design.ImageEditorFormBase.PerformApplyChanges()
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at InvokeStub_SendOrPostCallback.Invoke(Object, Span`1)
   at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

**Comment by gmhRTS** [comment]
This is definitely a blocker for me. Is there a previous version of System.Text.Json that I can use as a work around until this gets fixed?

**Comment by gmhRTS** [comment]
Actually, I downgraded to version 8.05 and all the dependencies for it to their respective versions of 8 as well which seemed to resolve the problem. I am on version 17.12.3 of Visual Studio 2022 Pro.

**Comment by Tanya-Solyanik** [comment]
Does adding https://www.nuget.org/packages/System.IO.Pipelines make any difference before the downgrade to 8.05?

**Comment by devcsantos** [comment]
> Does adding https://www.nuget.org/packages/System.IO.Pipelines make any difference before the downgrade to 8.05?

The issue is still reproducible even after adding the `System.IO.Pipelines` NuGet package to the project. For reference, the project used v9.0.0 for both `System.Text.Json` and `System.IO.Pipelines`.

**Comment by Olina-Zhang** [comment]
> [@Olina-Zhang](https://github.com/Olina-Zhang) can your team please test?

Can repro it in Visual Studio 17.12.3 build as above repro steps in DevExpress WinForms Project with .NET 8.0

https://github.com/user-attachments/assets/5da6a948-1188-46f9-a7eb-030082ab50f0

Failed to verify DevExpress WinForms Projec with .NET 7.0, because designer cannot be loaded after adding `System.Text.Json` 
![Image](https://github.com/user-attachments/assets/0c00d73f-e5cb-449b-b3bc-923c4d3b5416)

@Tanya-Solyanik it also repro after adding `System.IO.Pipelines` NuGet package with `System.Text.Json` together

**Comment by Tanya-Solyanik** [comment]
@Olina-Zhang  - could you please move this bug to the designer repo?

**Comment by Olina-Zhang** [comment]
Filed a designer issue: https://github.com/microsoft/winforms-designer/issues/6212.

@devcsantos @gmhRTS you can use `System.Text.Json` NuGet package with a lower version 8.0.5 for DevExpress WinForms .NET 8.0 Project, I tested it works well.
![Image](https://github.com/user-attachments/assets/9a097bc7-7654-429e-ba1c-63ca896fa32a)

**Comment by merriemcgaw** [comment]
I'm going to close this issue here, but we can update you when we have a resolution on our end.

**Comment by merriemcgaw** [comment]
@Shyam-Gupta is taking a look on our end.

**Comment by saitorhan** [comment]
@Olina-Zhang I installed new DevExpress version with Visual Studio 17.12.3 and got problem again.

**Comment by Olina-Zhang** [comment]
> [@Olina-Zhang](https://github.com/Olina-Zhang) I installed new DevExpress version with Visual Studio 17.12.3 and got problem again.

In my previous testing result:

- In released 17.12.3 + latest DevExpress WinForms Components (v24.2.3), this issue can repro in DevExpress WinForms .NET 8.0 Project, not DevExpress WinForms .NET 9.0 Project

- In preview 17.13 preview 2 + latest DevExpress WinForms Components (v24.2.3), everything works as expected.
